qa/workunits/rbd/rbd_mirror_ha.sh

```diff
@@ -0,0 +1,135 @@
+#!/bin/sh
+#
+# rbd_mirror_ha.sh - test rbd-mirror daemons in HA mode
+#
+
+. $(dirname $0)/rbd_mirror_helpers.sh
+
+wait_for_leader()
+{
+    local s instance
+
+    for s in 1 2 4 4 4 4 4 4; do
+	sleep $s
+	for instance in `seq 0 9`; do
+	    admin_daemon "${CLUSTER1}:${instance}" \
+			 rbd mirror status ${POOL} ${CLUSTER2} |
+		grep '"leader": true' || continue
+	    LEADER=${instance}
+	    return 0
+	done
+    done
+
+    return 1
+}
+
+test_replay()
+{
+    local image
+
+    for image; do
+	wait_for_image_replay_started ${CLUSTER1}:${LEADER} ${POOL} ${image}
+	write_image ${CLUSTER2} ${POOL} ${image} 100
+	wait_for_replay_complete ${CLUSTER1}:${LEADER} ${CLUSTER2} ${POOL} \
+				 ${image}
+	wait_for_status_in_pool_dir ${CLUSTER1} ${POOL} ${image} 'up+replaying' \
+				    'master_position'
+	if [ -z "${RBD_MIRROR_USE_RBD_MIRROR}" ]; then
+	    wait_for_status_in_pool_dir ${CLUSTER2} ${POOL} ${image} \
+					'down+unknown'
+	fi
+	compare_images ${POOL} ${image}
+    done
+}
+
+testlog "TEST: start the first daemon instance and test replay"
+start_mirror ${CLUSTER1}:0
+image1=test1
+create_image ${CLUSTER2} ${POOL} ${image1}
+LEADER=0
```

Comment:
Yes, as we already have LEADER as a global, mutated by wait_for_leader, I decided to use it in all functions.

Actually, in my previous version to avoid the global, I passed leader var to wait_for_leader by name (similarly to what we do in request_resync_image to set image_id), so it looked like:
```
wait_for_leader leader
test_replay ${leader} ${image1} ...
```
But this looked overkill to me here (I don't like very much using `eval` (to implement pass by name) and this would looked useful only if these were generic functions in helpers file). Still if you like this approach more I return to that version.